providers/src/airflow/providers/edge/cli/api_client.py

```diff
@@ -0,0 +1,120 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+import logging
+from datetime import datetime
+from http import HTTPStatus
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+from urllib.parse import quote, urljoin, urlparse
+
+import requests
+import tenacity
+from requests.exceptions import ConnectionError
+from urllib3.exceptions import NewConnectionError
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.providers.edge.worker_api.auth import jwt_signer
+from airflow.providers.edge.worker_api.datamodels import WorkerStateBody
+
+if TYPE_CHECKING:
+    from airflow.providers.edge.models.edge_worker import EdgeWorkerState
+
+logger = logging.getLogger(__name__)
+
+
+class AirflowHttpException(AirflowException):
```

Comment:
imp this should just inherit `HTTPException` from fastapi instead of a `AirflowException`

Reply:
Agree. We are overusing AirflowException

Reply:
This is client code and I would not like to bind this to FastAPI. One one hand it is independent and just a wrapper around the HTTP call to decide about if the call can be retried (see `_is_retryable_exception()` and on the other side Airflow 2.10 does not carry FastAPI - would like to prevent needing to add this dependency.

But as the Exception is just used within the module, the base exception already contains all fields... I propose to just drop the exception class and use the base class here.

Reply:
it can be from httpx too: https://www.python-httpx.org/exceptions/ -- doesn't have to be fastapi

Reply:
>I propose to just drop the exception class and use the base class here.

sure